test/extensions/filters/http/ext_proc/filter_test.cc

```diff
@@ -814,6 +814,7 @@ TEST_F(HttpFilterTest, PostAndChangeRequestBodyBuffered) {
 
   EXPECT_EQ(FilterHeadersStatus::StopIteration, filter_->decodeHeaders(request_headers_, false));
   processRequestHeaders(true, absl::nullopt);
+  EXPECT_EQ(request_headers_.getContentLengthValue(), "100");
 
   Buffer::OwnedImpl req_data;
   TestUtility::feedBufferWithRandomCharacters(req_data, 100);
```

Comment:
In below line 832, body is mutated. The test ext_proc server should add the header mutation with updated content-length in the response and send it back to ext_proc filter.

Reply:
That is existing test behavior and existing test code I don't touch.  This test is also called `PostAndChangeRequestBodyBuffered` so I don't think we need to process response.

The thing related to content-length you described is already tested integration test.